components/patient-measurements/api/src/main/java/org/phenotips/measurements/internal/AbstractMeasurementHandler.java

```diff
@@ -166,6 +199,51 @@ public void initialize() throws InitializationException
         this.chartConfigurations = this.settingsFactory.loadConfigurationsForMeasurementType(getName());
     }
 
+    @Override
+    public Collection<VocabularyTerm> getAssociatedTerms(Double standardDeviation)
+    {
+        ResourceBundle configuration = ResourceBundle.getBundle("measurementsAssociatedTerms");
+        List<VocabularyTerm> terms = new ArrayList<>();
+
+        if (null == standardDeviation || standardDeviation <= -3) {
+            addResolvedTermsToList(configuration, this.getName(), "extremeBelowNormal", terms);
+        }
+        if (null == standardDeviation || standardDeviation <= -2) {
```

Comment:
Thanks for the feedback, guys. Admittedly, I wrote this part of the code in a bit of a rush and not with too much thought towards style. I will go ahead with a refactor, doing my best to take all of this into account.